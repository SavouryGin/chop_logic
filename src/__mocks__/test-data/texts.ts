export const testText =
  'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Assumenda, hic illo. Qui alias quasi repudiandae dignissimos unde, id sit molestiae ducimus numquam nihil, sint impedit molestias perferendis ipsa nobis reiciendis!';
