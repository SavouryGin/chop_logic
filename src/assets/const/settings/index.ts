import { SelectEntity } from 'types';

export const languageOptions: SelectEntity[] = [
  { option: { en: 'English', ru: 'Английский' }, value: 'en' },
  { option: { en: 'Russian', ru: 'Русский' }, value: 'ru' },
];

export const externalLinks: { [key in string]: string } = {
  telegram: 'https://telegram.me/savoury_gin',
  gitHub: 'https://github.com/SavouryGin',
  mail: 'mailto:Dmitrii.Suroviagin@gmail.com',
};
