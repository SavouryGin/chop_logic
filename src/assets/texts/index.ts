export * from 'assets/texts/buttons';
export * from 'assets/texts/inputs';
export * from 'assets/texts/ui-elements';
