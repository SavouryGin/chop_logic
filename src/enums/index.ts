export enum Icon {
  Default = 'chop-icon__check',
  Home = 'chop-icon__home',
  Menu = 'chop-icon__menu',
  Enlarge = 'chop-icon__enlarge2',
  Shrink = 'chop-icon__shrink2',
  Settings = 'chop-icon__cog',
  Delete = 'chop-icon__bin2',
  Up = 'chop-icon__circle-up',
  Down = 'chop-icon__circle-down',
  Right = 'chop-icon__circle-right',
  Left = 'chop-icon__circle-left',
  Sound = 'chop-icon__volume-medium',
  NoSound = 'chop-icon__volume-mute2',
  LightMode = 'chop-icon__sun',
  DarkMode = 'chop-icon__contrast',
}

export enum Pages {
  Home,
  NotFound,
  Propositions,
  Predicates,
  TruthTables,
  Syllogisms,
}
