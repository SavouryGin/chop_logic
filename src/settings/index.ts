import { SelectEntity } from 'types';

export const languageOptions: SelectEntity[] = [
  { option: { en: 'English', ru: 'Английский' }, value: 'en' },
  { option: { en: 'Russian', ru: 'Русский' }, value: 'ru' },
];

export const languageStringOptions = ['en', 'ru'];

export const externalLinks: { [key in string]: string } = {
  telegram: 'https://telegram.me/savoury_gin',
  gitHub: 'https://github.com/SavouryGin',
  linkedIn: 'https://www.linkedin.com/in/dmitrii-suroviagin/',
  mail: 'mailto:Dmitrii.Suroviagin@gmail.com',
  facebook: 'https://www.facebook.com/dmitrii.suroviagin',
};
