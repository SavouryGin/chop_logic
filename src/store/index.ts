import createSagaMiddleware from 'redux-saga';
import rootReducer from './root-reducer';
import rootSaga from './root-saga';
import { configureStore } from '@reduxjs/toolkit';

const sagaMiddleware = createSagaMiddleware();

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat([sagaMiddleware]),
});

sagaMiddleware.run(rootSaga);
