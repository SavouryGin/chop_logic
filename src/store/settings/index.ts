import { Language } from 'types';
import { PayloadAction, createSlice } from '@reduxjs/toolkit';
import { SettingsFlag, SettingsInitialState } from './interfaces';

export const settingsInitialState: SettingsInitialState = {
  language: 'en',
  flags: {
    isNavigationOpen: false,
    isSidebarOpen: false,
    isDarkMode: false,
    isFullScreen: false,
    isSettingOpen: false,
    isSoundsEnabled: false,
  },
};

export const settingsSlice = createSlice({
  name: 'settings',
  initialState: settingsInitialState,
  reducers: {
    resetSettings: () => {
      return settingsInitialState;
    },

    toggleFlag: (state, action: PayloadAction<SettingsFlag>) => {
      const currentFlagValue = state.flags[action.payload];
      state.flags[action.payload] = !currentFlagValue;
    },

    setLanguage: (state, action: PayloadAction<Language>) => {
      state.language = action.payload;
    },

    setFullScreenFlag: (state, action: PayloadAction<boolean>) => {
      state.flags.isFullScreen = action.payload;
    },

    setUpFlag: (state, action: PayloadAction<{ flag: SettingsFlag; value: boolean }>) => {
      const { flag, value } = action.payload;
      state.flags[flag] = value;
    },
  },
});

export const settingsActions = { ...settingsSlice.actions };
