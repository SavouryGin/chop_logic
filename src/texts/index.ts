export * from './buttons';
export * from './inputs';
export * from './ui-elements';
export * from './propositions';
