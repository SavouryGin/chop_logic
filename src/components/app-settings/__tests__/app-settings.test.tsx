import AppSettings from '../index';
import React from 'react';
import renderWithRedux from 'utils/testing/render-with-redux';
import { ButtonID, InputID } from 'enums';
import { buttonTexts, inputTexts } from 'utils/texts';
import { combineReducers } from '@reduxjs/toolkit';
import { fireEvent, screen } from '@testing-library/react';
import { languageOptions } from 'utils/settings';
import { settingsInitialState, settingsSlice } from 'store/settings';

describe('AppSettings tests:', () => {
  const mockedReducer = combineReducers({ settings: settingsSlice.reducer });
  const mockedState = { settings: settingsInitialState };

  beforeEach(() => {
    renderWithRedux(<AppSettings className={'test-class'} />, mockedReducer, mockedState);
  });

  it('renders the settings container', () => {
    const container = screen.getByTestId('settings');
    expect(container).toBeInTheDocument();
    expect(container).toHaveClass('settings', 'test-class');
  });

  it('renders the form component', () => {
    expect(screen.getByRole('form')).toBeInTheDocument();
  });

  it('displays the language options', () => {
    const options = screen.getAllByRole('option');
    expect(options).toHaveLength(languageOptions.length);
    for (let i = 0; i < options.length; i++) {
      expect(options[i]).toHaveTextContent(languageOptions[i].option.en);
    }
  });

  it('displays the Dark Mode checkbox unchecked by default', () => {
    const checkbox = screen.getByLabelText(inputTexts[InputID.isDarkModeCheckbox].label.en);
    expect(checkbox).toBeInTheDocument();
    expect(checkbox).not.toBeChecked();
    fireEvent.click(checkbox);
    expect(checkbox).toBeChecked();
  });

  it('displays the Sounds checkbox unchecked by default', () => {
    const checkbox = screen.getByLabelText(inputTexts[InputID.isSoundsCheckbox].label.en);
    expect(checkbox).toBeInTheDocument();
    expect(checkbox).not.toBeChecked();
    fireEvent.click(checkbox);
    expect(checkbox).toBeChecked();
  });

  it('displays the apply button', () => {
    const applyBtn = screen.getByTitle(buttonTexts[ButtonID.Apply].title.en);
    expect(applyBtn).toHaveProperty('type', 'submit');
    expect(applyBtn).toHaveTextContent('Apply');
  });

  it('should match the snapshot', () => {
    const { asFragment } = renderWithRedux(<AppSettings className={'test-class'} />, mockedReducer, mockedState);
    expect(asFragment()).toMatchSnapshot();
  });
});
