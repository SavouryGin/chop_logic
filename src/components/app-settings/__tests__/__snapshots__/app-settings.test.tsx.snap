// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSettings tests: should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="settings test-class"
    data-testid="settings"
  >
    <form
      action="/"
      aria-label="form"
      class="form"
    >
      <div
        class="select"
      >
        <select
          class="select__field"
          id="select_id_1"
          name="language"
        >
          <option
            value="en"
          >
            English
          </option>
          <option
            value="ru"
          >
            Russian
          </option>
        </select>
        <label
          class="label select__label"
          data-testid="label-test-id"
          for="select_id_1"
        >
          <span
            class="label__text"
          >
            Language
          </span>
        </label>
      </div>
      <div
        class="checkbox-input"
      >
        <input
          class="checkbox-input__default"
          id="checkbox_id_2"
          name="isDarkMode"
          type="checkbox"
        />
        <label
          class="label checkbox-input__label chop-icon__checkbox-unchecked"
          data-testid="label-test-id"
          for="checkbox_id_2"
        >
          <span
            class="label__text"
          >
            Dark Mode
          </span>
        </label>
      </div>
      <div
        class="checkbox-input"
      >
        <input
          class="checkbox-input__default"
          id="checkbox_id_3"
          name="isSoundsEnabled"
          type="checkbox"
        />
        <label
          class="label checkbox-input__label chop-icon__checkbox-unchecked"
          data-testid="label-test-id"
          for="checkbox_id_3"
        >
          <span
            class="label__text"
          >
            Sounds
          </span>
        </label>
      </div>
      <button
        class="button button_large"
        id="button_id_7"
        title="Apply"
        type="submit"
      >
        <span
          class="button__shadow"
        />
        <span
          class="button__edge"
        />
        <span
          class="button__front chop-icon__check"
          data-testid="button_id_7"
        >
          Apply
        </span>
      </button>
    </form>
  </div>
</DocumentFragment>
`;
